Added picflip - @divpreeet

divpreeet opened the pull request.

JosiasAurel reviewed (changes requested):
How much time did you spend developing this project?

Also please remove the following folders
- `dist`
- `picflig.egg-info`
Remove the LICENSE and README.md file at the root of the `submissions` folder

divpreeet commented:
Hey! thanks a lot for the review,
> How much time did you spend developing this project?

The project was started long ago, and at the time I didn't have wakatime, but the time tracked by wakatime (in the past two days or so) is around 2 hours, but the time that wasn't tracked was an approx work of two hours or so.

 Also the `picflip-egg.info` and the `dist` folders are removed!

divpreeet commented:
Oops, didn't read the whole thing in a hurry,

also deleted the `license` and `README.md`

Sorry for the inconvenience!

JosiasAurel commented on submissions/requirements.txt:
put your requirements in the same folder as the source code for your project

divpreeet replied:
Updated.

JosiasAurel commented on submissions/setup.py:
put your setup.py in the same folder as the source code for your project

divpreeet replied:
Updated.

JosiasAurel reviewed (changes requested):
See comments

divpreeet commented:
I did the needed. create a folder, its now `submission - picflip/` and the `picflip/` stores the `piclip/requirements.txt, setup.py, __init__.py, cli.py`, and there is also a build folder!

JosiasAurel reviewed (changes requested):
This is what I get when I run `picflip` after running `pip install .` in the project
<img width="574" alt="image" src="https://github.com/user-attachments/assets/1d9c2738-dd27-46ea-a209-4706b5fb03b0" />

divpreeet commented:
try installing it from pypi, it seems to work on both my computers..

JosiasAurel reviewed (changes requested):
A few comments
- Please improve the help menu by showing examples on how to use each command
- This CLI could support converting more image file types. Example from PNG to JPG, JPG to JPEG etc. Allow specifying the output file format
- Please fix such that it's easier to install and run from the source code directly.

divpreeet commented:
Alright! here's what i've changed

1. Updated the help command to include examples. the help command now works with ```-h or -help```
2. Added functionality to convert images with ```pillow```, and it works for jpeg, jpg, webp, and png, for svgs, you have input only, an example command is ```python cli.py shrek_image.png shrek_image.jpeg jpeg```, so basically ```<input image> <output image> <image format>```
3. Updated the readme so the project is easier to run using source code, with step by step instructions
4. Also i found out why you received the error, ig it was due to the setup.py and requirements being in the same folder, so i updated the PR, and now the project structure is ```submissions/picflip/setup.py, requirements.txt, picflip/__init.py__, cli.py

lmk if any more changes are required!

JosiasAurel reviewed (approved):
Great! Please fill [this form](https://airtable.com/appzR6MIcj5G9A2Fa/pag8jn2axMXOB2lid/form) to get your grant.

divpreeet commented:
Already did it!

JosiasAurel commented:
> Already did it!

Good
